src/test/matching.test.js

```diff
@@ -0,0 +1,192 @@
+      longitude: parseFloat(8.845367431640627),
+    },
+  },
+  categories: {
+    0: true,
+    2: true,
+  },
+};
+const pinInfoCategory1 = {
+  userId: '123',
+  title: 'Pin_2',
+  area: {
+    radius: 5,
+    location: {
+      latitude: parseFloat(47.23563352505211),
+      longitude: parseFloat(8.845367431395730),
+    },
+  },
+  categories: {
+    1: true,
+  },
+};
+const pinInfoCategory0123 = {
+  userId: '573',
+  title: 'Pin_1',
+  area: {
+    radius: 5,
+    location: {
+      latitude: parseFloat(47.23563352505248),
+      longitude: parseFloat(8.845367431640627),
+    },
+  },
+  categories: {
+    0: true,
+    1: true,
+    2: true,
+    3: true,
+  },
+};
+const postInfoCategory1 = {
+  userId: authUser123.uid,
+  title: 'Farmer Market',
+  location: {
+    latitude: parseFloat(47.23563352505248),
+    longitude: parseFloat(8.845367431640627),
+  },
+  category: '1',
+};
+const postInfoCategory2 = {
+  userId: authUser123.uid,
+  title: 'Pub Tour',
+  location: {
+    latitude: parseFloat(47.23563352505211),
+    longitude: parseFloat(8.845367431395730),
+  },
+  category: '2',
+};
+const postInfoCategory0 = {
+  userId: authUser573.uid,
+  title: 'Bike-Tour',
+  location: {
+    latitude: parseFloat(47.23563352505248),
+    longitude: parseFloat(8.845367431640627),
+  },
+  category: '0',
+};
+const postInfoShouldntMatch = {
+  userId: authUser573.uid,
+  title: 'Hiking',
+  location: {
+    latitude: parseFloat(67.23563352505248),
+    longitude: parseFloat(9.845367431640627),
+  },
+  category: '0',
+};
+const pinStatus = { ready1: false, ready2: false, ready3: false };
+const postStatus = {
+  ready1: false, ready2: false, ready3: false, ready4: false,
+};
+
+const setUpForUse = () => {
+  createPin(
+    pinInfoCategory02, () => { pinStatus.ready1 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPin(
+    pinInfoCategory1, () => { pinStatus.ready2 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPin(
+    pinInfoCategory0123, () => { pinStatus.ready3 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPost(
+    postInfoCategory1, () => { postStatus.ready1 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPost(
+    postInfoCategory2, () => { postStatus.ready2 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPost(
+    postInfoCategory0, () => { postStatus.ready3 = true; },
+    (error) => { console.log('error:'); console.log(error); },
+  );
+  createPost(
+    postInfoShouldntMatch, () => { postStatus.ready4 = true; },
+    (error) => { console.log('error'); console.log(error); },
+  );
+};
+
+afterEach(() => {
+  deleteTestDbOnRootLevel();
+});
+
+beforeEach(() => {
+  haltIfLiveDB();
+  deleteTestDbOnRootLevel();
+  setUpForUse();
+});
+
+describe('Test matching', () => {
+  describe('#checks Listing', () => {
```

Comment:
Why is there a hashtag?

Reply:
because you did it in the first place

Reply:
That was not intentional. Can be deleted.